components/meter/meter-linear.js

```diff
@@ -32,23 +51,66 @@ class MeterLinear extends LitElement  {
 				position: absolute;
 				top: 0;
 			}
+
+			.d2l-meter-linear-text {
+				color: var(--d2l-color-ferrite);
+				display: flex;
+				flex-direction: row;
+				justify-content: space-between;
+				margin: 0 6px;
+			}
+
+			.d2l-meter-linear-secondary {
+				align-self: flex-end;
+			}
+
 		`];
 	}
 
 	constructor() {
 		super();
-		this.value = 0;
 		this.max = 100;
+		this.percent = false;
+		this.value = 0;
 	}
 
 	render() {
-		const progressBarWidth = this.value / this.max * 100;
+		const primaryTextId = getUniqueId();
+		const secondaryTextId = getUniqueId();
+
+		const percentage = this.max > 0 ? this.value / this.max * 100 : 0;
+		const primary = this.percent ? `${Math.floor(percentage)}%` : `${this.value}/${this.max}`;
+		const secondaryText = this._formatContext(this.text, this.value, this.max);
+		const secondaryTextElement = secondaryText ? html`<div id="${secondaryTextId}" class="d2l-meter-linear-secondary">${secondaryText}</div>` : html``;
 		return html `
-			<div class="d2l-meter-linear-full-bar">
-				<div class="d2l-meter-linear-inner-bar" style="width:${progressBarWidth}%;"></div>
+			<div
+				role="progressBar"
+				aria-valuenow="${this.value}"
+				aria-valuemax="${this.max}"
+				aria-valuetext="${ifDefined(secondaryText)}"
+				aria-describedby="${primaryTextId} ${secondaryTextId}">
+				<div class="d2l-meter-linear-full-bar">
+					<div class="d2l-meter-linear-inner-bar" style="width:${percentage}%;"></div>
+				</div>
+				<div class="d2l-body-small d2l-meter-linear-text">
+					<div id="${primaryTextId}" class="d2l-meter-linear-primary">${primary}&nbsp;</div>
+					${secondaryTextElement}
+				</div>
 			</div>
 		`;
 	}
+
+	_formatContext(context, value, max) {
+		if (!context) {
+			return;
+		}
+
+		// Floor so you don't get 100% when you are at 99.5%
+		context = context.replace('{%}', `${Math.floor(value / max * 100)}%`);
```

Comment:
When you want to format this percent correctly, you can use the localize mixin's `formatNumber`:

    formatNumber(0.189, {style: 'percent'});